airflow/providers/ssh/hooks/ssh.py

```diff
@@ -218,7 +218,8 @@ def get_conn(self) -> paramiko.SSHClient:
         else:
             if self.host_key is not None:
                 client_host_keys = client.get_host_keys()
-                client_host_keys.add(self.remote_host, 'ssh-rsa', self.host_key)
+                remote_host = f"[{self.remote_host}]:{self.port}" if self.port != SSH_PORT else self.remote_host
+                client_host_keys.add(remote_host, 'ssh-rsa', self.host_key)
```

Comment:
As I mentioned in the linked issue, it may be easier to do it this way:

```python
client_host_keys.add(self.remote_host, 'ssh-rsa', self.host_key)
if self.port:
    client_host_keys.add(f"{self.remote_host}:{self.port}", 'ssh-rsa', self.host_key)
```

Reply:
I'm not convinced it is a good idea to add the host key without port unconditionally. At least in theory it might be possible that there is one server listening on the default port and another one on self.port. Those servers could have different public keys.

However, it might be more clear to have that explicit branch instead of the inline if. I will adjust the PR accordingly.

Reply:
How about this?

```python
if self.port == SSH_PORT:
    client_host_keys.add(self.remote_host, 'ssh-rsa', self.host_key)
if self.port:
    client_host_keys.add(f"{self.remote_host}:{self.port}", 'ssh-rsa', self.host_key)
```

So SSH servers exposed on the default port can have both registered.

Reply:
In line 196 the port is set to SSH_PORT if it is None:

        self.port = self.port or SSH_PORT

Hence, checking for `self.port is None` is always false. Your suggestion results in adding both `self.remote_host` and `f"{[self.remote_host}]:{self.port}"` in most cases. Adding the latter for the standard port is not required according to the OpenSSH documentation (https://en.wikibooks.org/wiki/OpenSSH/Client_Configuration_Files#~/.ssh/known_hosts). As Paramiko expects the strings according to the OpenSSH format, we would just be adding a redundant, never used entry.

Reply:
I like how it is now actually. I am also all for adding host without port when the default port is used. This is what OpenSSH is doing usually.